### PR TITLE
Random select the subset of validation set in lp when validation set is huge

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit-tests/test_dataloading.py [hunk 1]
```diff
@@ -638,6 +638,23 @@ def test_GSgnnLinkPredictionTestDataLoader(batch_size, num_negative_edges):
...
+        num_samples = 0
+        for pos_neg_tuple, sample_type in dataloader:
+            num_samples += 1
+            assert isinstance(pos_neg_tuple, dict)
+            assert len(pos_neg_tuple) == 1
+            for _, pos_neg in pos_neg_tuple.items():
+                pos_src, _, pos_dst, _ = pos_neg
+                assert len(pos_src) <= batch_size
+
+        assert num_samples ==  -(-fixed_test_size // batch_size) * 2
```
Comment (zhjwy9343) [inline tests/unit-tests/test_dataloading.py:656]: Why use two negative signs, "-", here

**Review by zhjwy9343 — APPROVED** [review]
LGTM
